sympy/geometry/parabola.py

```diff
@@ -406,3 +407,21 @@ def vertex(self):
             vertex = Point(focus.args[0], focus.args[1] - self.p_parameter)
 
         return vertex
+from sympy import *
+def tangent(a,b,g,f,h,c,x1,y1):  
+    x,y=symbols('x y')
+    P=Poly((x**2)*a+(y**2)*b+2*g*x+2*h*x*y+2*f*y+c,(x,y))
+    S=Eq((x**2)*a+(y**2)*b+(2*g*x)+(2*h*x*y)+(2*f*y),-c)
+    S1=P(x1,y1)
+    if S1==0:
```

Comment:
Please add spaces around the commas and operators. For instance `def tangent(a, b, g, f, h, c, x1, y1): ` and  `S1 = P(x1, y1)`. Apply wherever it is necessary